MekHQ/src/mekhq/gui/dialog/CustomizePersonDialog.java

```diff
@@ -295,16 +307,89 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
         });
         gridBagConstraints = new java.awt.GridBagConstraints();
         gridBagConstraints.gridx = 1;
-        gridBagConstraints.gridy = 3;
+        gridBagConstraints.gridy = y;
         gridBagConstraints.gridwidth = 1;
         gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
         gridBagConstraints.fill = java.awt.GridBagConstraints.HORIZONTAL;
         gridBagConstraints.insets = new java.awt.Insets(5, 5, 0, 0);
         panDemog.add(choiceGender, gridBagConstraints);
 
+        y++;
+
         gridBagConstraints = new java.awt.GridBagConstraints();
         gridBagConstraints.gridx = 0;
-        gridBagConstraints.gridy = 4;
+        gridBagConstraints.gridy = y;
+        gridBagConstraints.anchor = java.awt.GridBagConstraints.WEST;
+        gridBagConstraints.insets = new java.awt.Insets(0, 5, 0, 0);
+        panDemog.add(new JLabel("Origin Faction:"), gridBagConstraints);
+
+        DefaultComboBoxModel<Faction> factionsModel = new DefaultComboBoxModel<Faction>();
+        List<Faction> orderedFactions = Faction.getFactions().stream()
+            .sorted(new FactionSorter(person.getCampaign()))
+            .collect(Collectors.toList());
+        for (Faction faction : orderedFactions) {
```

Comment:
Is it possible to refactor some of this logic into separate worker functions? This method is kind of large as is.